etrago/cluster/gas.py

```diff
@@ -960,14 +977,24 @@ def run_spatial_clustering_gas(self):
     ValueError: If the selected method is not "kmeans" or "kmedoids-dijkstra".
 
     """
-    if "CH4" in self.network.buses.carrier.values:
+
+    if ("CH4" in self.network.buses.carrier.values) | (
+        "H2_grid" in self.network.buses.carrier.values
+    ):
         settings = self.args["network_clustering"]
+        scn = self.args["scn_name"]
 
         if settings["active"]:
             method = settings["method_gas"]
             logger.info(f"Start {method} clustering GAS")
 
-            gas_network, weight, n_clusters = preprocessing(self)
+            ch4_network, weight_ch4, n_clusters_ch4 = preprocessing(
+                self, "CH4"
+            )
+            if scn not in ["eGon2035", "status2019"]:

```

Comment:
Do you think there is a more generic way for this, like checking if there are H2-pipelines? 
If it stays like this we have to check for each new scenario if it is correct.

Reply:
Change done in a9bdfe01c2e285637fb9d539d3d484f4e4d98b45